dracut/30afterburn/afterburn-hostname.service

```diff
@@ -0,0 +1,24 @@
+[Unit]
+Description=Afterburn Hostname
+# Azure needs to fetch the hostname in the initramfs
+ConditionKernelCommandLine=|ignition.platform.id=azure
+
+# This will be firstboot only to set up the correct
+# hostname for kube.
+ConditionFirstBoot=yes
```

Comment:
One question, does this actually work correctly in the initrd? For the real root, systemd keys off of `/etc/machine-id`, but here it has to be able to make that determination before `/sysroot` is even mounted. So does it key off of the *initrd*'s `/etc/machine-id`? Except that check will never change result unless we regen the initrd.

An alternative here is to hook into `ignition-complete.target`, which was made for this: https://github.com/coreos/ignition-dracut/blob/df88988f2f0791ca9ea1e14298c8523501d980c5/dracut/30ignition/ignition-complete.target#L3.

For RHCOS, since it doesn't have that target yet, you could do e.g. `ignition-files.service.requires`?

Reply:
Afterburn does not depend on Ignition though, so we probably shouldn't make it use that target. I agree though that using `ConditionFirstBoot` from the initramfs seems sketchy. Not sure the best path forward here. I think this is a case where the distro-dependence is showing.

Reply:
> Afterburn does not depend on Ignition though, so we probably shouldn't make it use that target.

We explicitly reference ignition targets, I don't really see a harm in cross-linking per se? Unless you want them to be as separate as possible.

I did some testing with a metal-bios image. If I `rd.break` into the initramfs on the first boot, I see the unit as inactive, and a message that: `Condition: start condition failed` (and same with subsequent boots). Interesting if I try to `systemctl start` the process manually, it will add a line `CondtionFirstBoot=yes was not met`. I'm not sure if that's the expected behaviour or not (i.e. which condition is not being met).

In either case, this does not seem to be very robust... should we

Reply:
> Afterburn does not depend on Ignition though, so we probably shouldn't make it use that target.

I'd say this unit is inherently dependent on Ignition already by simply having `ConditionKernelCommandLine=|ignition.platform.id=azure`. :)

We could do something fancy here and "connect" the two in the overlay. Though unless there's demand for running Afterburn without Ignition, I'm not sure if it's worth the investment.

Reply:
I think this is fine now that I think about it. Part of me still just wants one giant fcos-dracut repo, but I think that ship has sailed.